js/atoms/src/SelectSource.svelte

```diff
@@ -1,22 +1,30 @@
 <script lang="ts">
-	import { Microphone, Upload, Video } from "@gradio/icons";
+	import { Microphone, Upload, Webcam, ImagePaste } from "@gradio/icons";
 
-	export let sources: string[];
-	export let active_source: string;
+	type sources = "upload" | "microphone" | "webcam" | "clipboard" | null;
+
+	export let sources: Partial<sources>[];
```

Comment:
why the use of Partial here and throughout this file? I don't believe it makes a difference in this context.
```suggestion
	export let sources: sources[];
```

Reply:
also would name the type differently if we also have a variable called sources.